operator/src/main/java/com/github/streamshub/console/dependents/ConsoleSecret.java

```diff
@@ -110,6 +130,119 @@ protected Secret desired(Console primary, Context<Console> context) {
+        Map<Class<?>, List<?>> deploymentResources = new HashMap<>();
+
+        for (var metricsSource : Optional.ofNullable(primary.getSpec().getMetricsSources())
+                .orElse(Collections.emptyList())) {
+            var truststore = metricsSource.getTrustStore();
+
+            if (truststore != null) {
+                reconcileTrustStore(primary, context, data, metricsSource.getName(), METRICS_TRUST_PREFIX, truststore, "metrics-source", deploymentResources);
+            }
+        }
+
+        for (var registry : Optional.ofNullable(primary.getSpec().getSchemaRegistries())
+                .orElse(Collections.emptyList())) {
+            var truststore = registry.getTrustStore();
+
+            if (truststore != null) {
+                reconcileTrustStore(primary, context, data, registry.getName(), REGISTRY_TRUST_PREFIX, truststore, "schema-registry", deploymentResources);
+            }
+        }
+
+        context.managedDependentResourceContext().put("TrustStoreResources", deploymentResources);
+    }
+
+    @SuppressWarnings("java:S107") // Ignore Sonar warning for too many args
+    private void reconcileTrustStore(
+            Console primary,
+            Context<Console> context,
+            Map<String, String> data,
+            String sourceName,
+            String sourcePrefix,
+            TrustStore truststore,
+            String bucketPrefix,
+            Map<Class<?>, List<?>> deploymentResources) {
+
+        String namespace = primary.getMetadata().getNamespace();
+        String secretName = instanceName(primary);
+        String typeCode = truststore.getType().toString();
+        String volumeName = replaceNonAlphanumeric(sourcePrefix + sourceName, '-');
+
+        @SuppressWarnings("unchecked")
+        List<Volume> volumes = (List<Volume>) deploymentResources.computeIfAbsent(Volume.class, k -> new ArrayList<>());
+
+        volumes.add(new VolumeBuilder()
+                .withName(volumeName)
+                .withNewSecret()
+                    .withSecretName(secretName)
+                    .addNewItem()
+                        .withKey(sourcePrefix + sourceName + ".content")
+                        .withPath(sourcePrefix + sourceName + "." + typeCode)
+                    .endItem()
+                    .withDefaultMode(420)
+                .endSecret()
+                .build());
+
+        @SuppressWarnings("unchecked")
+        List<VolumeMount> mounts = (List<VolumeMount>) deploymentResources.computeIfAbsent(VolumeMount.class, k -> new ArrayList<>());
+
+        mounts.add(new VolumeMountBuilder()
+                .withName(volumeName)
+                .withMountPath("/etc/ssl/" + sourcePrefix + sourceName + "." + typeCode)
+                .withSubPath(sourcePrefix + sourceName + "." + typeCode)
+                .build());
+
+        String configTemplate = "quarkus.tls.\"" + bucketPrefix + "-%s\".trust-store.%s.%s";
```

Comment:
Ultimately, I'd like to avoid configuration of these buckets this way and have it supported via the console's YAML file, but that will be a bit too involved for the initial support I think.